jdbc-sink/src/main/java/org/springframework/cloud/stream/module/jdbc/JdbcSinkConfiguration.java

```diff
@@ -0,0 +1,217 @@
+						MapSqlParameterSource parameterSource = new MapSqlParameterSource();
+						for (String key: columnExpressionVariations.keySet()) {
+							List<Expression> spels = columnExpressionVariations.get(key);
+							Object value = NOT_SET;
+							EvaluationException lastException = null;
+							for (Expression spel : spels) {
+								try {
+									value = spel.getValue(evaluationContext, message);
+								}
+								catch (EvaluationException e) {
+									lastException = e;
+								}
+							}
+							if (value == NOT_SET) {
+								if (lastException != null) {
+									logger.info("Could not find value for column '" + key + "': " + lastException.getMessage());
+								}
+								parameterSource.addValue(key, null);
+							}
+							else {
+								if (value instanceof JsonPropertyAccessor.ToStringFriendlyJsonNode) {
+									// Need to do some reflection until we have a getter for the Node
+									DirectFieldAccessor dfa = new DirectFieldAccessor(value);
+									JsonNode node = (JsonNode) dfa.getPropertyValue("node");
+									Object valueToUse;
+									if (node == null || node.isNull()) {
+										valueToUse = null;
+									}
+									else if (node.isNumber()) {
+										valueToUse = node.numberValue();
+									}
+									else if (node.isBoolean()) {
+										valueToUse = node.booleanValue();
+									}
+									else {
+										valueToUse = node.textValue();
+									}
+									parameterSource.addValue(key, valueToUse);
+								}
+								else {
+									parameterSource.addValue(key, value);
+								}
+							}
+						}
+						return parameterSource;
+					}
+				});
+		return jdbcMessageHandler;
+	}
+
+	@ServiceActivator(autoStartup = "false", inputChannel = Sink.INPUT)
+	public void handleMessage(Message<?> msg) {
+		msgHandler.handleMessage(msg);
+	}
+
+	@ConditionalOnProperty("initialize")
+	@Bean
+	public DataSourceInitializer nonBootDataSourceInitializer(DataSource dataSource, ResourceLoader resourceLoader) {
+		DataSourceInitializer dataSourceInitializer = new DataSourceInitializer();
+		dataSourceInitializer.setDataSource(dataSource);
+		ResourceDatabasePopulator databasePopulator = new ResourceDatabasePopulator();
+		databasePopulator.setIgnoreFailedDrops(true);
+		dataSourceInitializer.setDatabasePopulator(databasePopulator);
+		if ("true".equals(properties.getInitialize())) {
+			databasePopulator.addScript(new DefaultInitializationScriptResource(properties));
+		} else {
+			databasePopulator.addScript(resourceLoader.getResource(properties.getInitialize()));
+		}
+		return dataSourceInitializer;
+	}
+
+	@Bean
+	public JdbcOperations jdbcOperations(DataSource dataSource) {
+		return new JdbcTemplate(dataSource);
+	}
+
+	/*
+	 * This is needed to prevent a circular dependency issue with the creation of the converter.
+	 */
+	public static class Nested {
+		@Bean
+		@ConfigurationPropertiesBinding
+		public ShorthandMapConverter shorthandMapConverter() {
```

Comment:
I wonder how is it resolved if we don't have here `@Configuration` or even `@Component` for `light` `@Bean`?..